src/Microsoft.ML.Data/Prediction/CalibratorCatalog.cs

```diff
@@ -186,7 +188,9 @@ private protected CalibratorTransformer(IHostEnvironment env, ModelLoadContext c
 
             // *** Binary format ***
             // model: _calibrator
+            // scoreColumnName: _scoreColumnName
             ctx.LoadModel<TICalibrator, SignatureLoadModel>(env, out _calibrator, "Calibrator");
+            _scoreColumnName = ctx.LoadString();
```

Comment:
I think this will fail if we load Calibrators that we serialized before your changes, because they won't include the new `scoreColumnName` string. This needs to be fixed by optionally reading the new string, depending on the value on `ctx.Header.ModelVerWritten`. E.g. in the same way we load text loaders:

https://github.com/dotnet/machinelearning/blob/709ec020b702fe69e0dcbf87fbcb538bea0af7b7/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L1459-L1469

Adding a backward compatibility test with _at least_ one kind of calibrator would also be useful to ensure they are loaded correctly :wink:

Reply:
Added!